.github/workflows/test.yml

```diff
@@ -73,3 +73,7 @@ jobs:
       - name: Run tests
         run: |
           make test-no-log
+      - name: Upload coverage report to codecov
+        uses: codecov/codecov-action@v3
+        with:
+          files: ./coverage/coverage.json
```

Comment:
do we need additional steps to attach codecov integration in our PRs?
something like https://github.com/apps/codecov